### PR TITLE
DEFRA-ENV01-Emissions-of-air-pollutants

### PR DESCRIPTION
**Pull request by osamede20**
https://github.com/GSS-Cogs/family-climate-change/issues/6

Transform done in line with detailed spec

### CODE REVIEW

File: datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+distribution = metadata.distribution(latest=True)
+
+tabs = distribution.as_databaker()
+print(len(tabs), ' ', [tab.name for tab in tabs])
+
+contents = tabs[1]
+titles = contents.excel_ref('B9').expand(DOWN) - contents.excel_ref('B22').expand(DOWN)
+title_list = [title.value for title in titles]
+print(*title_list, sep ='\n')
+print(len(title_list))
```
Comment (Shannon95) [inline datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py:38]: print statements can be removed in final code - great for stage one though (I need to get in the better practice of removing them)

File: datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+trace = TransformTrace()
```
Comment (Shannon95) [inline datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py:40]: Are we still using the trace function?

File: datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                    HDimConst('Measure Type', "% of total emissions")
+            ]
+        tidi_sheet_2 = ConversionSegment(tab, dimensions2, observations2)
+        table2 = tidi_sheet_2.topandas()
+        trace.store('dataframe' + tab.name + str(j), table2)
+        df_percent = trace.combine_and_trace(title2, 'dataframe' + tab.name + str(j)).fillna('')
+        df_percent['Period'] = df_percent['Period'].str.replace('\.0', '')
+        df_percent.rename(columns={'OBS':'Value'}, inplace=True)
+        df_percent['Value'] = pd.to_numeric(df_percent['Value'], downcast='float')
+        df_percent['Value'] = df_percent['Value'].round(2)
```
Comment (Shannon95) [inline datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py:198]: Should we be rounding the percentage or values in general? Something to maybe ask DM

**Review by Shannon95 — COMMENTED** [review]
Overall looks good for stage one transform. Just a few comments to consider; 
- I don't believe we are using the trace function anymore, though I think the decision was to just leave it in those transforms that already have it in place. 
- Multiple cube outputs that I believe can be joined together to become one.

**Comment by osamede20** [comment]
Thanks Shannon.

File: datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py (hunk 2 above)
Comment (osamede20) [reply datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py:31]: I think we can use the trace function if we choose to, but without the commenting functionality.
I find it quite useful as it shows where the datacubes for the tabs are joined.

**Pull request merged**